### ISSUE TITLE
Simple way to work with Eigen col based Sparse matrices

### ISSUE THREAD

**Issue by mobereng** [ddemidov/amgcl#183]
Hi Denis !

I'm currently using the library and I ran into multiple problems pretty much related with parsing the data from my matrix A and send it to the library.

At the moment I ended up by creating the CRS components by hand, reading the entire Eigen::Sparse matrix, which is clearly not practical, I can get the result but I have to parse in the matrix A and also the vector solution to a VectorXd. 

So my question is, is there any way better to do this ? I also saw that the library contains Matrices Adapters, like `amgcl/adapter/eigen.hpp` but I'm not sure this will work regarding the fact my matrix is col based not row based

To sum up, the data I'm working with is Eigen based and I expect the following as data structures:
`Eigen::Sparse A;` \
`Eigen::VectorXd b;` \
`Eigen::VectorXd x;` \
Where x is the result vector

Appreciate your help

**Comment by ddemidov** [comment]
This example should help you:
https://gist.github.com/ddemidov/7894e9f89ae0f56a39d4a39599db5166

**Comment by ddemidov** [comment]
Sorry, I missed the part about the matrices being col-based. Amgcl does not support that, you need row-based matrices.

**Comment by mobereng** [comment]
Perfect ! Let me find a way to make my matrices row based, adjust a little bit the problem, thanks for the help !

**Comment by mobereng** [comment]
I did the following as last resource for being able to use my col based Eigen matrices

```c++
typedef SparseMatrix<double> SparseMatrixXd;
typedef SparseMatrix<double, Eigen::RowMajor> SparseMatrixXdRowMajor;

SparseMatrixXdRowMajor to_rowmajor(
  const SparseMatrixXd & A
) {
  SparseMatrixXdRowMajor bridge(A.rows(), A.rows());
  VectorTriplet triplets;
  for (int k = 0; k < A.outerSize(); ++k)
    for (SparseMatrixXd::InnerIterator it(A, k); it; ++it)
      triplets.push_back(TripletDouble((int)it.row(), (int)it.col(), it.value()));
  bridge.setFromTriplets(triplets.begin(), triplets.end());
  return bridge;
}
```

I think it will impact a bit the performance but was the best I coud do to make it compatible. Also implement the `AMGCL_USE_EIGEN_VECTORS_WITH_BUILTIN_BACKEND()` as you mention, so now I can work directly with the Eigen format

Thanks for the help,
And it would be nice in a future have a direct interpretation of col based eigen matrices

**Comment by ddemidov** [comment]
Does not something like this work for you?
```cpp
Eigen::SparseMatrix<double, Eigen::ColMajor> A(3,3);
A.insert(0,2) = 1;
A.insert(1,2) = 2;
A.makeCompressed();

Eigen::SparseMatrix<double, Eigen::RowMajor> B = A;
std::cout << A.coeff(0,2) << " " << A.coeff(1,2) << std::endl;
std::cout << B.coeff(0,2) << " " << B.coeff(1,2) << std::endl;
```
